lms/djangoapps/grades/models.py

```diff
@@ -47,12 +47,17 @@ def to_json(self):
         if self._json is None:
             sorted_blocks = sorted(self, key=attrgetter('locator'))
             list_of_block_dicts = [block._asdict() for block in sorted_blocks]
+            course_key = next(iter(self)).locator.course_key  # all blocks are from the same course
```

Comment:
Thinking about it, I'm pretty sure we explicitly decided to save it for empty subsections as a "negative cache," so we wouldn't have to hit the db just to figure out that something was empty.